src/Compilers/CSharp/Test/Emit/CodeGen/PatternTests.cs

```diff
@@ -5914,33 +5914,32 @@ .locals init (bool V_0)
       IL_001c:  ldc.i4.0
       IL_001d:  stloc.0
       IL_001e:  ldloc.0
-      IL_001f:  brtrue.s   IL_0024
-      IL_0021:  ldc.i4.0
-      IL_0022:  br.s       IL_0025
-      IL_0024:  ldc.i4.1
-      IL_0025:  ret
+      IL_001f:  ret
     }
 ");
             compVerifier.VerifyIL("C.M2", @"
     {
-      // Code size       25 (0x19)
+      // Code size       30 (0x1e)
```

Comment:
>// Code size       30 (0x1e) [](http://example.com/codeflow?start=6&length=28)

Perhaps we shouldn't activate the optimization if the condition is going to involve short-circuiting branches anyway. Be a little bit more conservative.

Reply:
I changed the logic to only optimize if there's a comparison operator.